File: app/src/main/java/com/tripplanner/di/component/AppComponent.java
package com.tripplanner.di.component;

import android.app.Application;

import com.tripplanner.BaseApp;
import com.tripplanner.di.builder.ActivityBuilder;
import com.tripplanner.di.module.AppModule;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

/**
 * Created by suyashg on 09/06/18.
 */

@Singleton
@Component(modules = {AndroidInjectionModule.class, AppModule.class, ActivityBuilder.class})
public interface AppComponent {

    void inject(BaseApp app);

    @Component.Builder
    interface Builder {

        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }
}

File: kotlinapp/src/main/java/com/tripplanner/di/AppComponent.kt
package com.tripplanner.di

import com.tripplanner.BaseApp
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

/**
 * Created by suyashg on 21/06/18.
 */
@Singleton
@Component(
        modules = [
        AndroidSupportInjectionModule::class,
        ActivityBuilder::class,
        AppModule::class
        ])
interface AppComponent : AndroidInjector<BaseApp> {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: BaseApp): Builder

        fun network(network: NetworkModule): Builder

        fun build(): AppComponent
    }

    override fun inject(app: BaseApp)

}